### DISCUSSION TITLE
How to update Table data source from pandas dataframe by button click?

### DISCUSSION THREAD

**Discussion by SyniLSN** [zauberzeug/nicegui#2253] [Q&A]
### Description

I created a table and wanted the button to execute pandas. read_excel Assign df to table, but update or use table.from_ Pandas cannot meet the requirements, as shown in the following example:

```python
df = pd.DataFrame()
def get_erp_stock():
    df = pd.read_excel("export.xlsx")
    mytable.from_pandas(df) # not work
    mytable.update() # not work


test = ui.button(text="test", on_click=get_erp_stock)
ui.page_title("test")
mytable = ui.table.from_pandas(df).style("width:100%;height:100%;")

ui.run()

```

If I use `mytable = ui.table.from_pandas(df).style("width:100%;height:100%;") ` to create a table and then use `mytable.from_pandas(df)` in method `get_erp_stock()`, it will result in creating a new table below this button without updating data in the original table; If mytable. update() is used, neither a new table will be created nor data will be updated in the original table. I need to synchronize the data in the table after clicking the button and updating the df, similar to the data binding in Vue or WPF.

**Comment by MattMacs** [comment]
Hi,

you are not calling get_erp_stock, replace `on_click=test` with `on_click=get_erp_stock` and it should work.

**Reply by SyniLSN** [reply]
Sorry, I made a low-level error in the code here, which I have corrected and added a more detailed description.

**Comment by falkoschindler** [comment]
`from_pandas` is a builder method which creates a new table element. This is not what you want.

But you can try something like this:
```py
mytable.columns = [{'name': col, 'label': col, 'field': col} for col in df.columns]
mytable.rows = df.to_dict('records')
```
This should update the existing table with new data.
